utils/threshAlgos.py

```diff
@@ -1187,8 +1209,16 @@ def sbitRateAnalysis(chamber_config, rateTree, cutOffRate=0.0, debug=False, outf
     # Get the data
     import root_numpy as rp
     import numpy as np
-    list_bNames = ['dacValX','link','nameX','rate','shelf','slot','vfatCH','vfatN']
 
+    #for backwards compatibility, handle input trees that do not have a detName branch by finding the detName using the chamber_config
+    hasDetNameBranch = False
+    if 'detName' in rateTree.GetListOfBranches():
+        hasDetNameBranch = True
+
+    list_bNames = ['dacValX','link','nameX','rate','shelf','slot','vfatCH','vfatN','detName']
+    if not hasDetNameBranch:
+        list_bNames.remove('detName')
+        
```

Comment:
Can this block be abstracted in the several places it is used?

Reply:
I think this is done only twice and in separate in separate files.

Reply:
OK, for now I guess it's fine, but I'd consider consolidating this in the future